src/components/menu/menu.tsx

```diff
@@ -150,28 +179,181 @@ export class Menu {
+                        text: '',
+                        icon: {
+                            name: 'home',
+                        },
+                        type: 'icon-only',
+                    },
+                    ...breadCrumbItems,
+                ]}
+            />
+        );
+    };
+
+    private renderMenuList = () => {
+        const items = this.visibleItems;
+
+        return (
+            <limel-menu-list
+                tabindex="1"
+                style={{
+                    'overflow-y': 'auto',
+                    'flex-grow': '1',
+                }}
+                class={{
+                    'has-grid-layout has-interactive-items': this.gridLayout,
+                }}
+                items={items}
+                type="menu"
+                badgeIcons={this.badgeIcons}
+                onSelect={this.onSelect}
+                ref={this.setListElement}
+                onKeyDown={this.handleMenuKeyDown}
+            />
+        );
+    };
+
+    /**
+     * Key handler for the menu list
+     * Can go forward/back with righ/left arrow keys
+     * @param {KeyboardEvent} event event
+     * @returns {void}
+     */
+    private handleMenuKeyDown = (event: KeyboardEvent) => {
+        const isLeft =
+            event.key === ARROW_LEFT || event.keyCode === ARROW_LEFT_KEY_CODE;
+
+        const isRight =
+            event.key === ARROW_RIGHT || event.keyCode === ARROW_RIGHT_KEY_CODE;
+
+        if (!isLeft && !isRight) {
+            return;
+        }
+
+        if (!this.gridLayout) {
+            const currentItem = this.getCurrentItem();
+
+            event.stopPropagation();
+            event.preventDefault();
+            if (isRight) {
+                this.goForward(currentItem);
+            } else if (isLeft) {
+                this.goBack(currentItem);
+            }
+        }
+    };
+
+    private getCurrentItem = (): MenuItem => {
+        const activeItem = this.list?.shadowRoot?.querySelector(
+            '[role="menuitem"][tabindex="0"]'
+        );
+        const attrIndex = activeItem?.attributes?.getNamedItem('data-index');
+        const dataIndex = parseInt(attrIndex?.value || '0', 10);
+
+        return this.visibleItems[dataIndex] as MenuItem;
+    };
+
+    private goForward = (currentItem: MenuItem) => {
+        this.handleSelect(currentItem, false);
+    };
+
+    private goBack = (currentItem: MenuItem) => {
+        const parentItem = currentItem?.parentItem;
+        if (!parentItem) {
+            // Already in the root of the menu
+            return;
+        }
+
+        const grandParent = parentItem.parentItem;
+        if (!grandParent) {
+            // Is only one step down, go to the root of the menu.
+            // No need to load a sub-menu.
+            this.currentSubMenu = null;
+            this.currentItems = null;
+            this.navigateMenu.emit(null);
+
+            return;
+        }
+
+        this.handleSelect(grandParent);
+    };
```

Comment:
I don't understand why we're going to "grandParent" here? If we want to navigate one step up from "current", surely that must be "parent" then? 🤔

Reply:
Ah, is it because we we're not going to the parent, we're selecting the grandparent item, which den loads the parent? Is that also why there is loading of items when moving outwards from a sub-menu to its parent menu (in the lazy-loading example in the next PR)?

Reply:
Yes, exactly. We "simulate" a click on the grandParent to go to the parent so to say.